crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -254,11 +254,27 @@ private ObjectNode buildSpec(Crd.Spec crd, Class<? extends CustomResource> crdCl
             result.set("additionalPrinterColumns", cols);
         }
         if (crd.subresources().status().length != 0) {
-            ObjectNode statusNode = nf.objectNode();
+            ObjectNode subresources = nf.objectNode();
+
             if (crd.subresources().status().length > 0) {
-                statusNode.set("status", nf.objectNode());
+                subresources.set("status", nf.objectNode());
+            }
+
+            if (crd.subresources().scale().length > 0) {
```

Comment:
In the [CRD schema](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#customresourcesubresources-v1beta1-apiextensions-k8s-io) `status` and `scale` are both scalar quantities. When I added support for the `status` subresource wanted to have a way to say "this CRD supports status" or not, and I chose not to use a boolean flag for that, but rather use the presence/absence of the `@Status` annotation to indicate that. Since you can only omit an annotation value if it has a default, and you can't default an annotation to null that forced me to make `status` an array. You're using the same pattern here, which I guess is fine, but I suppose we should have the `CrdGenerator` exit with an error if the length of the array is greater than 1.

Reply:
That makes sense. I added a check and if both `status` or `scale` have length > 1 it throws RuntimeException which should stop it.